corehq/apps/app_manager/views/apps.py

```diff
@@ -277,12 +277,6 @@ def app_source(request, domain, app_id):
 
 
 @require_can_edit_apps
-def copy_app_check_domain(request, domain, name, app_id_or_source):
```

Comment:
Thanks for investigating further (and for telling me about `-S`, which I did not know or had forgotten about). Definitely agree it's an important check. Having `copy_app_check_domain` as a separate view when it's never called as a view seems misleading, so I just added a commit to put back the domain-checking logic but keep it inside `copy_app`. I think it's clearer, lmk if you find it more confusing than the original.